client/packages/system/src/Name/Components/DonorSearchInput/DonorSearchInput.tsx

```diff
@@ -1,45 +1,47 @@
-import React, { FC } from 'react';
-import {
-  Autocomplete,
-  useBufferState,
-  useTranslation,
-} from '@openmsupply-client/common';
+import React from 'react';
+import { Autocomplete, useTranslation } from '@openmsupply-client/common';
 import { NameRowFragment, useName } from '../../api';
-import {
-  basicFilterOptions,
-  filterByNameAndCode,
-  NameSearchInputProps,
-} from '../../utils';
+import { basicFilterOptions, filterByNameAndCode } from '../../utils';
 import { getNameOptionRenderer } from '../NameOptionRenderer';
 
-export const DonorSearchInput: FC<NameSearchInputProps> = ({
+export interface DonorSearchInputProps {
+  onChange: (donor: NameRowFragment | null) => void;
+  width?: number;
+  donorId: string | null;
+  disabled?: boolean;
+  clearable?: boolean;
+}
+
+export const DonorSearchInput = ({
   onChange,
-  onInputChange,
   width = 250,
-  value,
+  donorId,
   disabled = false,
   clearable = false,
-}) => {
+}: DonorSearchInputProps) => {
   const { data, isLoading } = useName.document.donors();
-  const [buffer, setBuffer] = useBufferState(value);
```

Comment:
I'm not sure where the buffer state came from, tested in the existing usage (equipment summary) and was fine without... probably a copy paste from another name input